File: String rp or pr[22-March-2023]/BruteForce(c++ and java)/Solution.cpp
//			>> C++ CODE <<


class Solution {
public:
    long long recUtil(int X, int Y, string S) {
        if (S.length() == 0) {
            return 0;
        }

        long long max_val = 0;

        for (int i = 0; i < S.length() - 1; i++) {
            if (S[i] == 'r' && S[i + 1] == 'p') {
                string str = S.substr(0, i) + S.substr(i + 2, S.length());
                max_val = max(max_val, Y + recUtil(X, Y, str));
            }
            else if (S[i] == 'p' && S[i + 1] == 'r') {
                string str = S.substr(0, i) + S.substr(i + 2, S.length());
                max_val = max(max_val, X + recUtil(X, Y, str));
            }
        }

        return max_val;
    }

    long long solve(int X, int Y, string S) {
        return recUtil(X, Y, S);
    }
};

File: String rp or pr[22-March-2023]/BruteForce(c++ and java)/Solution.java
//				>> JAVA CODE <<

class Solution 
{ 
    static long recUtil(int X,int Y,String S)
    {
        if(S.length()==0) return 0;
        
        long max=0;
        
        for(int i=0;i<S.length()-1;i++)
        {
            if(S.charAt(i)=='r' && S.charAt(i+1)=='p')
            {
                String str=S.substring(0,i)+S.substring(i+2,S.length());
                max=Math.max(max,Y+recUtil(X,Y,str));
            }
            else if(S.charAt(i)=='p' && S.charAt(i+1)=='r')
            {
                String str=S.substring(0,i)+S.substring(i+2,S.length());
                max=Math.max(max,X+recUtil(X,Y,str));
            }
        }
        
            return max;
    }
    static long solve(int X,int Y, String S)
	{    
	    return recUtil(X,Y,S);
	}
} 